music21/stream/base.py

```diff
@@ -10661,7 +10648,7 @@ def _consolidateLayering(self, layeringMap):
                 for k in post:
                     # this comparison needs to be based on object id, not
                     # matching equality
-                    if id(elementObj) in [id(e) for e in post[k]]:
+                    if any(True for e in post[k] if e is elementObj):
```

Comment:
icky (`any(True...`)-- but definitely faster when elementObj is close to the beginning, so keep your new version.